Added custom user model with customer user manager

bartczak-pa opened the pull request:
This overwrites default django behaviour and allows to create users based on email instead of username.

Can we check if this code is okay?

bartczak-pa committed:
Added custom user model with customer user manager

This overwrites default django behaviour and allows to create users based on email instead of username

jacoor commented:
@bartczak-pa please see tests results. Those failed.
https://github.com/pymasterspl/reddit/actions/runs/9045129435/job/24854541143 
If you look here, it failed on migrations. This will be checked only after all is green. Please fix migrations.

bartczak-pa committed:
Added custom user model with customer user manager

This overwrites default django behaviour and allows to create users based on email instead of username

bartczak-pa committed:
Added custom user model with customer user manager

This overwrites default django behaviour and allows to create users based on email instead of username

bartczak-pa commented:
@jacoor Migrations are fixed and code is passing tests now.

jacoor committed:
add D102 to ignore: remove docstring required

jacoor committed:
Type fixes example

jacoor reviewed (changes requested):
The code looks good.
However, Ruff is configured on purpose to enforce typings. 
Please, do fix ruff errors and avoid #noqa unless really necessary. 
I pushed an example of type updates for you, please fix other missing types. 
If you have issues with typing, you can try to use AI to fix 
- copilot - ask it to add typings for you (this works)
- or copy the code to Chat GPT and ask to add types.
Of course, do not forget to use `ruff check --fix` for other issues. 
Again, only use #noqa if this is really necessary, like:
`users/models.py:29:19: TRY003 Avoid specifying long messages outside the exception class`, however, for this one google has very good answer you can find after couple seconds search:
https://docs.astral.sh/ruff/rules/raise-vanilla-args/
(I put "how to fix TRY003 Avoid specifying long messages outside the exception class`" as a question to google).

jacoor committed:
Removed one of noqa's

jacoor committed:
fix for extrafields type

jacoor commented on users/models.py:
```diff
@@ -1,3 +1,41 @@
+class UserManager(BaseUserManager):
+    use_in_migrations: bool = True
+
+    def _create_user(self: "UserManager", email: str, password: str, **extra_fields: int) -> "User":
```
Here you have an example of added typings.

jacoor commented on users/models.py:
```diff
@@ -1,3 +1,41 @@
+
+    def _create_user(self: "UserManager", email: str, password: str, **extra_fields: int) -> "User":
+        if not email:
+            message: str = "Users must have an email address"
```
And fixed ruff errors, this one is EM101

bartczak-pa committed:
Added custom user model with customer user manager

This overwrites default django behaviour and allows to create users based on email instead of username

jacoor commented:
@bartczak-pa you need to resolve conflicts. After that, tests will start to run and we will know if the code good. 
Let me know if you need help.

bartczak-pa committed:
Merge branch 'dev' into feature/users_module

jacoor commented:
Great work. Now merge away!

bartczak-pa commented:
Typings and conflicts are resolved.
